source/security/firmware-resiliency-and-recovery.rst

```diff
@@ -0,0 +1,96 @@
+            +------------------------------------------------------------------------+
+            |   KEYH   |  0x5ff000(0xFF7FF000)  |  0x001000  |  Uncompressed, R_A    |
+            |   CNFG   |  0x5fb000(0xFF7FB000)  |  0x004000  |  Uncompressed, R_A    |
+            |   FWUP   |  0x5db000(0xFF7DB000)  |  0x020000  |  Compressed  , R_A    |
+            |   SG02   |  0x519000(0xFF719000)  |  0x0c2000  |  Compressed  , R_A    |
+            +------------------------------------------------------------------------+
+            |                              REDUNDANT B                               |
+            +------------------------------------------------------------------------+
+            |   KEYH   |  0x518000(0xFF718000)  |  0x001000  |  Uncompressed, R_B    |
+            |   CNFG   |  0x514000(0xFF714000)  |  0x004000  |  Uncompressed, R_B    |
+            |   FWUP   |  0x4f4000(0xFF6F4000)  |  0x020000  |  Compressed  , R_B    |
+            |   SG02   |  0x432000(0xFF632000)  |  0x0c2000  |  Compressed  , R_B    |
+            +------------------------------------------------------------------------+
+            |                             NON REDUNDANT                              |
+            +------------------------------------------------------------------------+
+            |   PYLD   |  0x402000(0xFF602000)  |  0x030000  |  Compressed  ,  NR    |
+            |   UVAR   |  0x401000(0xFF601000)  |  0x001000  |  Uncompressed,  NR    |
+            |   EPLD   |  0x2a0000(0xFF4A0000)  |  0x161000  |  Uncompressed,  NR    |
+            |   MRCD   |  0x290000(0xFF490000)  |  0x010000  |  Uncompressed,  NR    |
+            |   VARS   |  0x28e000(0xFF48E000)  |  0x002000  |  Uncompressed,  NR    |
+            |   IPFW   |  0x28d000(0xFF48D000)  |  0x001000  |  Uncompressed,  NR    |
+            |   EMTY   |  0x072000(0xFF272000)  |  0x21b000  |  Uncompressed,  NR    |
+            +------------------------------------------------------------------------+
+            |                              NON VOLATILE                              |
+            +------------------------------------------------------------------------+
+            |   RSVD   |  0x071000(0xFF271000)  |  0x001000  |  Uncompressed,  NV    |
+            |   EMTY   |  0x000000(0xFF200000)  |  0x071000  |  Uncompressed,  NV    |
+            +----------+------------------------+------------+-----------------------+
+
+Whenever a failure is detected on one partition, a boot is tried from the other partition. If successful, the failing partition is overwritten with the working partition. Then,
+a normal boot to OS is executed, cancelling out a firmware update if one is in flight. Thus, this can occur during a normal boot, or a update-related boot. So, the primary partition
+can be copied over to the backup partition and vice versa.
+
+The diagram below shows the boot flow when a boot failure occurs on the primary boot partition.

```

Comment:
We can add a block diagram for the resiliency feature and later go to the flow charts for detailed view.